massa-bootstrap/src/listener.rs

```diff
@@ -77,19 +78,33 @@ impl BootstrapTcpListener {
 }
 
 impl BSEventPoller for BootstrapTcpListener {
-    fn poll(&mut self) -> Result<PollEvent, BootstrapError> {
+    fn poll(&mut self) -> Result<Vec<PollEvent>, BootstrapError> {
         self.poll.poll(&mut self.events, None).unwrap();
 
         // Confirm that we are not being signalled to shut down
         if self.events.iter().any(|ev| ev.token() == STOP_LISTENER) {
-            return Ok(PollEvent::Stop);
+            return Ok(vec![PollEvent::Stop]);
         }
 
-        // Ther could be more than one connection ready, but we want to re-check for the stop
-        // signal after processing each connection.
-        Ok(PollEvent::NewConnection(
-            self.server.accept().map_err(BootstrapError::from)?,
-        ))
+        let mut results = Vec::new();
+
+        // Process each event.
+        for event in self.events.iter() {
+            match event.token() {
+                NEW_CONNECTION => {
+                    results.push(PollEvent::NewConnection(self.server.accept()?));
+                }
+                _ => unreachable!(),
+            }
+        }
+
+        // We need to have an accept() error with WouldBlock, otherwise polling may not raise any new events.
+        // See https://users.rust-lang.org/t/why-mio-poll-only-receives-the-very-first-event/87501
+        if self._mio_server.accept().is_ok() {
+            debug!("Mio server still had bootstrap connection data to read");
```

Comment:
What are your thoughts on escalating this to error?

Reply:
I like it.

how about this:

```rust
while let Ok(conn) = self._mio_server.accept() {
    warn!("...");
    results.push(...);
}
```
...My thinking being is from the comment in the linked conversation:

> Therefore, you must keep calling accept until it fails with a WouldBlock error.

...and we won't be throwing away the connection either.

Thoughts?

Reply:
The problem with that all the other code handles std::net::TcpStream, and mio_server.accept() yields mio::net::TcpStream.
It may be the cleanest way to do it (I don't really like that we have both an std listener and a mio listener on the same socket), but it's probably a bigger (and maybe unwanted for other reasons) refactor.

Reply:
...it was a bit of a neccisary hack. 

I can see why you chose not to go that way now.